stingray/crossspectrum.py

```diff
@@ -1566,6 +1566,18 @@ def _initialize_from_any_input(
         ``crossspectrum_from_XXXX`` function, and initialize ``self`` with
         the correct attributes.
         """
+        if segment_size is None and isinstance(data1, StingrayObject):
+            common_gti = cross_two_gtis(data1.gti, data2.gti)
+            data1.gti = common_gti
+            data2.gti = common_gti
+            data1 = data1.apply_gtis(inplace=False)
+            data2 = data2.apply_gtis(inplace=False)
+
+            if hasattr(data1, "counts"):
```

Comment:
Why is this only checked if `data1` has an attribute `counts`? Could that not also be the case for a time series that does *not* have counts?

Reply:
Sure, you have a different behavior in that case though. The presence of counts tells you that this is coming from binned events

Reply:
I specified that this has to be an array attribute, so, a light curve